test/testcafe/enLeaks/EnglishLeaks_spec.js

```diff
@@ -0,0 +1,170 @@
+  'authenticator-enroll-email.json',
+  'authenticator-enroll-data-phone.json',
+  'authenticator-enroll-data-phone-voice.json',
+  'terminal-enduser-email-consent-denied.json',
+  'terminal-return-email-consent-denied.json',
+  'terminal-return-email-consent.json',
+  'email-challenge-consent.json',
+  'error-authenticator-enroll-idp.json',
+  'error-authenticator-verification-idp.json',
+  'device-code-activate.json',
+  'error-invalid-device-code.json',
+  'terminal-device-activated.json',
+  'terminal-device-not-activated.json',
+];
+
+const parseMockData = () => {
+  // parse mocks folder
+  const mocks = [];
+  // eslint-disable-next-line no-console
+  console.log('================= Parsing mocks for en leaks automation =============');
+  fs.readdirSync(mocksFolder).forEach(file => {
+    const isIgnored = ignoredMocks.includes(file);
+    if (!isIgnored) {
+      mocks.push(file);
+    } else {
+      test.skip(`Warning skipping mock ${file} from test english leaks test suite. This file may result in english leaks on the UI.`,() => {});
+    }
+  });
+  return mocks;
+};
+
+const setUpResponse = (filePath) => {
+  const mockResponse = require(filePath);
+  // Majority of the mock files can be loaded with just mocking the introspect API and providing a response
+  // In some cases we may need to mock more than just introspect API to load the screen (example polling)
+  // For those cases add the url and responses to responseMap
+  const responseMap = [
+    { 
+      'url': 'http://localhost:3000/idp/idx/introspect',
+      'response': mockResponse
+    },
+    { 
+      'url': 'http://localhost:3000/idp/idx/challenge/poll',
+      'response': mockResponse
+    },
+  ];
+
+  const mock = RequestMock();
+  for (let i = 0; i < responseMap.length; i++) {
+    const url = responseMap[i].url;
+    const response = responseMap[i].response;
+    mock
+      .onRequestTo(url)
+      .respond(response);
+  }
+  return mock;
+};
+
+async function setup(t, locale) {
+  const widgetView = new PageObject(t);
+  await widgetView.navigateToPage();
+  await renderWidget({
+    'language': locale
+  });
+}
+
+const testEnglishLeaks = (mockIdxResponse, fileName, locale) => {
+  test.requestHooks(mockIdxResponse)(`${fileName} should not have english leaks`, async t => {
+    await setup(t, locale);
+    const viewText = await Selector('#okta-sign-in').textContent;
+    const noTranslationContentExists = await Selector('.no-translate').exists;
+    const noTranslationContent = noTranslationContentExists && await Selector('.no-translate').textContent;
+    await assertNoEnglishLeaks(fileName, viewText, noTranslationContent);
+  });
+};
+
+const runTest = (mockData) => {
+  for (let i = 0; i < mockData.length; i++) {
+    const fileName = mockData[i];
+    const filePath = `${mocksFolder}/${fileName}`;
+    // start test
+    testEnglishLeaks(setUpResponse(filePath), fileName, 'ok-PL');
+  }
+};
+
+const mockData = parseMockData();
+runTest(mockData);
+
+
+
```

Comment:
Curious - are these extra newlines due to a plugin? Usually there is just a single newline at the EOF.

Reply:
I think my vs code preferences did not get copied over when I switched laptops